Breeding-Insight/bi-api
# BI-2043 - Improve Observation Overwrite Performance

# Description
**Story:** [BI-2043](https://breedinginsight.atlassian.net/browse/BI-2043)

`ExperimentProcessor::initNewBrapiData` was calling `fetchOrCreateObservationPIO` in a nested loop for each phenotype column for each row, which in turn was calling `fetchExistingObservations`, an expensive method which resulted in 2 network calls to the backing BrAPI service. I was able to pull the `fetchExistingObservations` call outside the loops without changing the behavior of the code (except the BrAPI calls are now made just once). Performance is greatly improved. 

# Dependencies
bi-web: release/0.9
bi-api: bug/BI-2043

# Testing
>Feel free to use my test files attached below.
1. Upload an Experiment and Observations file with lots of phenotype columns and observations.
2. Download the experiment you just uploaded (all environments, any filetype).
3. Edit the file you downloaded, change the Observation values for a good number of phenotypes (tip: Type a new value in the top cell of a phenotype column and drag down from the corner to the bottom of your file to quickly overwrite the whole column).
4. Upload the edited file, make sure DeltaBreed displays the preview within 1 minute. Proceed with the overwrite, and ensure the data is saved correctly.


# Checklist:
